static/src/js/components/CollectionSelector/CollectionSelector.jsx

```diff
@@ -0,0 +1,374 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState
+} from 'react'
+
+import PropTypes from 'prop-types'
+
+import Badge from 'react-bootstrap/Badge'
+import Col from 'react-bootstrap/Col'
+import Form from 'react-bootstrap/Form'
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
+import Popover from 'react-bootstrap/Popover'
+import Row from 'react-bootstrap/Row'
+
+import {
+  FaMinus,
+  FaPlus,
+  FaTrash
+} from 'react-icons/fa'
+
+import { useLazyQuery, useSuspenseQuery } from '@apollo/client'
+
+import { GET_PERMISSION_COLLECTIONS } from '@/js/operations/queries/getPermissionCollections'
+
+import './CollectionSelector.scss'
+import { debounce, isEmpty } from 'lodash-es'
+
+import Button from '../Button/Button'
+/**
+ * @typedef {Object} CollectionSelectorComponentProps
+ * @property {Function} onChange A callback function triggered when the user selects collections.
+ * @property {Object} formData An Object with the saved metadata
+ */
+/**
+ * Renders a CollectionSelectorComponent component
+ *
+ * @component
+ * @example <caption>Render a CollectionSelectorComponent</caption>
+ * return (
+ *   <CollectionSelectorComponent />
+ * )
+ */
+const CollectionSelector = ({ onChange, formData }) => {
+  const [selected, setSelected] = useState([])
+
+  const [searchAvailable, setSearchAvailable] = useState('')
+  const [searchSelected, setSearchSelected] = useState('')
+
+  const [selectedAvailable, setSelectedAvailable] = useState([])
+  const [selectedSelected, setSelectedSelected] = useState([])
+
+  const { data: collectionList } = useSuspenseQuery(GET_PERMISSION_COLLECTIONS)
+
+  const { collections } = collectionList
+  const { items } = collections
+
+  const [available, setAvailable] = useState(items)
+
+  useEffect(() => {
+    if (!isEmpty(formData)) {
+      setSelected(formData)
+    }
+  }, [formData])
+
+  const moveToSelected = () => {
+    const newSelected = selectedAvailable.filter(
+      (item) => !selected.some((s) => s.conceptId === item.conceptId)
```

Comment:
Don't use single letter variable names (throughout this file). Using descriptive variable names makes the code easier to read and understand.